pkg/otlp/model/translator/dimensions.go

```diff
@@ -84,7 +84,29 @@ func (d *Dimensions) AddTags(tags ...string) *Dimensions {
 
 // WithAttributeMap creates a new metricDimensions struct with additional tags from attributes.
 func (d *Dimensions) WithAttributeMap(labels pcommon.Map) *Dimensions {
-	return d.AddTags(getTags(labels)...)
+	d2 := &Dimensions{
+		name:     d.name,
+		tags:     make([]string, 0, labels.Len()+len(d.tags)),
+		host:     d.host,
+		originID: d.originID,
+	}
+
+	collisionCheck := map[string]string{}
+	labels.Range(func(k string, v pcommon.Value) bool {
+		collisionCheck[k] = v.AsString()
+		d2.tags = append(d2.tags, utils.FormatKeyValueTag(k, v.AsString()))
+		return true
+	})
+
+	for _, tag := range d.tags {
+		sp := strings.Split(tag, ":")
+		if v, ok := collisionCheck[sp[0]]; ok && v != strings.Join(sp[1:], ":") {
+			d2.tags = append(d2.tags, "resource."+tag)
```

Comment:
Why are we assuming that _any_ duplicate that is a resource attribute and prefixing it as such? If we want to prefix all resource attributes like this shouldn't it be done at a different level where we are actually aware of this fact? It's a bit of a wild assumption.

Reply:
I don't think such a tradeoff is acceptable. I think we should instead simply ensure that the environment is taken from the right tag.

Reply:
Feel free to fix this however you would like. I put a few solid days of discovery into this issue and fix, so I'm tapped out, but I won't be offended if you go another route. Either way, just know that the potential for duplicate tags will still be a fundamental issue with the way the DataDog otlp adapter, and thereby the otel DD exporter, is currently designed.